src/metabase/lib/schema/common.cljc

```diff
@@ -46,9 +42,20 @@
   [:fn
    {:error/message "valid base type"
     :error/fn      (fn [{:keys [value]} _]
-                (str "Not a valid base type: " value))}
+                     (str "Not a valid base type: " (pr-str value)))}
    base-type?])
 
+(mr/def ::options
+  [:map
+   [:lib/uuid ::uuid]
+   ;; these options aren't required for any clause in particular, but if they're present they must follow these schemas.
+   [:base-type      {:optional true} [:maybe ::base-type]]
+   [:effective-type {:optional true} [:maybe ::base-type]]
+   [:semantic-type  {:optional true} [:maybe ::semantic-type]]
+   [:database-type  {:optional true} [:maybe ::non-blank-string]]
+   [:name           {:optional true} [:maybe ::non-blank-string]]
+   [:display-name   {:optional true} [:maybe ::non-blank-string]]])
```

Comment:
I added various things like `:base-type`, `:effective-type`, `:semantic-type`, etc. as optional keys to the lib options schema, so if those keys are present they have to conform to the relevant schemas.